wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/ServiceTerminator.java

```diff
@@ -30,12 +38,14 @@
   private static final Logger LOG = LoggerFactory.getLogger(ServiceTerminator.class);
 
   private final EventService eventService;
-  private final Set<ServiceTermination> terminations;
+  private final List<ServiceTermination> terminations;
 
   @Inject
   ServiceTerminator(EventService eventService, Set<ServiceTermination> terminations) {
     this.eventService = eventService;
-    this.terminations = terminations;
+    ArrayList<ServiceTermination> terminationsList = new ArrayList<>(terminations);
+    Collections.sort(terminationsList, new ServiceTerminationComparator(terminations));
+    this.terminations = terminationsList;
```

Comment:
can we do that in one shot? With stream for example?

Reply:
+1 consider to do it like `this.terminations = ImmutableSortedSet.copyOf(new Comparator<..>(), terminations);`